CMakeLists.txt

```diff
@@ -51,8 +52,22 @@ if(${enable_cbmc_tests})
     enable_testing()
 endif()
 
+option(use_smt_switch "use generic smt interface")
+if(use_smt_switch)
+  download_project(
+    PROJ smt-switch
+    GIT_REPOSITORY  https://github.com/thk123/smt-switch.git
+    GIT_TAG patch-1
+    )
+  add_subdirectory(${smt-switch_SOURCE_DIR})
+  add_compile_options(-DHAVE_SMT_SWITCH)
+#  if(BUILD_CVC4)
+    execute_process(COMMAND ./contrib/setup-cvc4.sh WORKING_DIRECTORY ${smt-switch_SOURCE_DIR})
+#  endif()
```

Comment:
I will add sections for each of the supported modules here, but might be worth having a think about whether we want one on by default, if you use smt-switch).